lib/rubocop/cop/style/multiple_comparison.rb

```diff
@@ -61,8 +61,13 @@ def on_new_investigation
 
         def on_or(node)
           root_of_or_node = root_of_or_node(node)
-
           return unless node == root_of_or_node
+
+          if or_root_changed?(root_of_or_node)
+            reset_comparison
+            @current_or_root = root_of_or_node
+          end
```

Comment:
We need to reset the context every time we find a new OR root expression, not just when we start a new investigation. Otherwise multiple short OR expressions add up into a false offense 😬